node/cn/peer_set_test.go

```diff
@@ -0,0 +1,533 @@
+	assert.NoError(t, peerSet.Register(enPeer))
+
+	assert.False(t, peerSet.closed)
+	peerSet.Close()
+	assert.True(t, peerSet.closed)
+}
+
+func TestPeerSet_SampleResendPeersByType_PN(t *testing.T) {
+	// CN Peer=1, PN Peer=0
+	{
+		peerSet := newPeerSet()
+		mockCtrl := gomock.NewController(t)
+
+		cnPeer := NewMockPeer(mockCtrl)
+		cnPeer.EXPECT().ConnType().Return(p2p.ConnType(node.CONSENSUSNODE)).Times(2)
+		setMockPeers([]*MockPeer{cnPeer})
+
+		assert.Equal(t, []Peer{}, peerSet.SampleResendPeersByType(node.PROXYNODE))
+		assert.NoError(t, peerSet.Register(cnPeer))
+		assert.Equal(t, []Peer{cnPeer}, peerSet.SampleResendPeersByType(node.PROXYNODE))
+
+		mockCtrl.Finish()
+	}
+	// CN Peer=0, PN Peer=1
+	{
+		peerSet := newPeerSet()
+		mockCtrl := gomock.NewController(t)
+
+		pnPeer := NewMockPeer(mockCtrl)
+		pnPeer.EXPECT().ConnType().Return(p2p.ConnType(node.PROXYNODE)).Times(3)
+		setMockPeers([]*MockPeer{pnPeer})
+
+		assert.Equal(t, []Peer{}, peerSet.SampleResendPeersByType(node.PROXYNODE))
+		assert.NoError(t, peerSet.Register(pnPeer))
+		assert.Equal(t, []Peer{pnPeer}, peerSet.SampleResendPeersByType(node.PROXYNODE))
+
+		mockCtrl.Finish()
+	}
+	// CN Peer=1, PN Peer=1
+	{
+		peerSet := newPeerSet()
+		mockCtrl := gomock.NewController(t)
+
+		cnPeer := NewMockPeer(mockCtrl)
+		pnPeer := NewMockPeer(mockCtrl)
+
+		cnPeer.EXPECT().ConnType().Return(p2p.ConnType(node.CONSENSUSNODE)).AnyTimes()
+		pnPeer.EXPECT().ConnType().Return(p2p.ConnType(node.PROXYNODE)).AnyTimes()
+
+		setMockPeers([]*MockPeer{cnPeer, pnPeer})
+
+		assert.Equal(t, []Peer{}, peerSet.SampleResendPeersByType(node.PROXYNODE))
+		assert.NoError(t, peerSet.Register(cnPeer))
+		assert.NoError(t, peerSet.Register(pnPeer))
+		assert.Equal(t, []Peer{cnPeer}, peerSet.SampleResendPeersByType(node.PROXYNODE))
+
+		mockCtrl.Finish()
+	}
+	// CN Peer=3, PN Peer=1
+	{
+		peerSet := newPeerSet()
+		mockCtrl := gomock.NewController(t)
+
+		cnPeer1 := NewMockPeer(mockCtrl)
+		cnPeer2 := NewMockPeer(mockCtrl)
+		cnPeer3 := NewMockPeer(mockCtrl)
+		pnPeer := NewMockPeer(mockCtrl)
+
+		cnPeer1.EXPECT().ConnType().Return(p2p.ConnType(node.CONSENSUSNODE)).AnyTimes()
+		cnPeer2.EXPECT().ConnType().Return(p2p.ConnType(node.CONSENSUSNODE)).AnyTimes()
+		cnPeer3.EXPECT().ConnType().Return(p2p.ConnType(node.CONSENSUSNODE)).AnyTimes()
+		pnPeer.EXPECT().ConnType().Return(p2p.ConnType(node.PROXYNODE)).AnyTimes()
+
+		setMockPeers([]*MockPeer{cnPeer1, cnPeer2, cnPeer3, pnPeer})
+
+		assert.Equal(t, []Peer{}, peerSet.SampleResendPeersByType(node.PROXYNODE))
+		assert.NoError(t, peerSet.Register(cnPeer1))
+		assert.NoError(t, peerSet.Register(cnPeer2))
+		assert.NoError(t, peerSet.Register(cnPeer3))
+		assert.NoError(t, peerSet.Register(pnPeer))
+		resendPeers := peerSet.SampleResendPeersByType(node.PROXYNODE)
+
+		assert.Equal(t, 2, len(resendPeers))
+		assert.False(t, containsPeer(pnPeer, resendPeers))
+
+		mockCtrl.Finish()
+	}
+}
+
+func TestPeerSet_SampleResendPeersByType_EN(t *testing.T) {
```

Comment:
How about testing SampleResendPeersByType_CN or _XX as an exception case for full coverage of `SampleResendPeersByType` function?

Reply:
What kind of exception case do you want me to cover? Can you suggest one?

Reply:
I meant considering the default case in following code.
```
func (pm *ProtocolManager) sampleResendPeersByType(nodetype p2p.ConnType) []Peer {
	// TODO-Klaytn Need to tune pickSize. Currently use 2 for availability and efficiency.
	var peers []Peer
	switch nodetype {
	case node.ENDPOINTNODE:
		peers = pm.peers.TypePeers(node.PROXYNODE)
		if len(peers) == 0 {
			peers = pm.peers.TypePeers(node.ENDPOINTNODE)
		}
		peers = samplingPeers(peers, 2)
	case node.PROXYNODE:
		peers = pm.peers.TypePeers(node.CONSENSUSNODE)
		if len(peers) == 0 {
			peers = pm.peers.TypePeers(node.PROXYNODE)
		}
		peers = samplingPeers(peers, 2)
	default:
		logger.Warn("Not supported nodetype", "nodetype", nodetype)
		return nil
	}
	return peers
}
```

Reply:
@KimKyungup Oh, I understood. As there are still uncovered lines in peer_set.go, how about adding a tc for the default case in another PR?

Reply:
Sure. :) Thanks.